cabal-ghc921.project

```diff
@@ -11,20 +11,44 @@ packages:
         --  ./plugins/hls-stylish-haskell-plugin
         --  ./plugins/hls-fourmolu-plugin
          ./plugins/hls-class-plugin
-         ./plugins/hls-eval-plugin
+        -- ./plugins/hls-eval-plugin
          ./plugins/hls-explicit-imports-plugin
          ./plugins/hls-refine-imports-plugin
-         ./plugins/hls-hlint-plugin
+        -- ./plugins/hls-hlint-plugin
+        ./plugins/hls-rename-plugin
         -- ./plugins/hls-retrie-plugin
-         ./plugins/hls-haddock-comments-plugin
-        --  ./plugins/hls-splice-plugin
+        -- ./plugins/hls-haddock-comments-plugin
+        -- ./plugins/hls-splice-plugin
          ./plugins/hls-qualify-imported-names-plugin
-         ./plugins/hls-floskell-plugin
+        -- ./plugins/hls-floskell-plugin
          ./plugins/hls-pragmas-plugin
          ./plugins/hls-module-name-plugin
-         ./plugins/hls-ormolu-plugin
+        -- ./plugins/hls-ormolu-plugin
          ./plugins/hls-call-hierarchy-plugin
-         ./plugins/hls-alternate-number-format-plugin
+        -- ./plugins/hls-alternate-number-format-plugin
+
+source-repository-package
+  type: git
+  location: https://github.com/tfausak/unix-compat
+  tag: 154c3a63f154cb49c51d5f9d13488e8119631d8a
+  -- To fix windows build
+  -- https://github.com/jacobstanley/unix-compat/pull/47
+
+source-repository-package
+  type: git
+  location: https://github.com/Bodigrim/th-extras
+  tag: f00ebca78f474d271fd7989cae0a0a47559b2efd
+  -- https://github.com/mokus0/th-extras/pull/14
+
+repository head.hackage.ghc.haskell.org
```

Comment:
Are we okay with merging this while still relying on head.hackage? I guess it's probably okay.

Reply:
yeah, if we got to build a working executable in the repo, we can offer it, although the hackage support will take more time (like happened with 9.0.1)

Reply:
Do we trust all the patches in `head.hackage` enough to ship binaries to our users based on them? Probably yes, but worth thinking about. (The other patches pulled in by `source-repository-package`s I assume got at least a brief lookover by whoever added them, but `head.hackage` has quite a bit of stuff.)

Reply:
`head.hackage` is of course an additional point of updates & failure. To remember to remove *asap*, it would be good to create a comment & report with the details of when to remove it.

Reply:
`head.hackage` is curated by a small set of GHC contributors, and in my view it's fully trusted

Reply:
It could be better to pinpoint a couple of `source-repository-package` with patches than depend on entire head.hackage. It will make lagging dependencies more visible, easier to track and push. Are there many packages left unupdated by now?

Reply:
This is a good suggestion, thanks @Bodigrim  
Do we have a volunteer to work on this? I won't be able to do it myself

Reply:
The question here for volunteers is how to distinguish what packages should have those patches. Is `buildable` criteria generally enough?

Reply:
On current last commig (7ab6b97ab036d76e913cf0de6910e66f5f11bc9c) I got:
```
[__5] trying: ghc-typelits-knownnat-0.7.6 (dependency of ghcide *test)
[__6] trying: ghc-typelits-natnormalise-0.7.6 (dependency of
ghc-typelits-knownnat)
[__7] next goal: base (dependency of ghcide)
[__7] rejecting: base-4.16.0.0/installed-4.16.0.0 (conflict:
ghc-typelits-natnormalise => ghc-bignum>=1.0 && <1.1, base =>
ghc-bignum==1.2/installed-1.2)
```
[Looks like the same situation from a nearby thread](https://github.com/haskell/haskell-language-server/pull/2503#discussion_r779593399), but now in 9.2.1.
```
allow-newer:
    *:ghc-bignum
```
not solved it.

`ghc-typelits-natnormalise` bound `ghc-bignum == 1.0.*`.

Opening a report in `ghc-typelits-natnormalise` to support new library versions.

Reply:
Would be reopened as a report after the merge into mainline.